src/main/java/seedu/address/logic/commands/SortWeaknessCommand.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Sort the list of persons in the address book by total weaknesses in descending order and display to the user.
+ */
+public class SortWeaknessCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort-weakness";
+
+    public static final String MESSAGE_SUCCESS = "Sorted all persons by most weaknesses";
+
+    /**
+     * Comparator that compares two Person objects by their total weaknesses.
+     * If total weaknesses of two Person objects are equal, then compare by least total strengths
+     */
+    protected static final Comparator<Person> BY_WEAKNESSES = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            //if total weaknesses are equal, order by least strengths
+            if (o1.getWeaknesses().size() == o2.getWeaknesses().size()) {
+                return o1.getStrengths().size() - o2.getStrengths().size();
+            } else if (o1.getWeaknesses().size() > o2.getWeaknesses().size()) {
+                return -1;
+            } else {
+                return 1;
+            }
```

Comment:
Probably can be cleaned up to
`else {
    return o2.getWeaknesses().size().compareTo(o1.getWeaknesses().size())
}`

Reply:
I kept the method in this format for the sake of readability. If the benefit doesn't outweigh the drawback, I'll consider reformatting these comparators and adding more comments. Thanks for bringing this up 👍